npm/private/npm_translate_lock.bzl

```diff
@@ -416,14 +416,19 @@ def _gen_npm_imports(lockfile, root_package, attr, registries, default_registry)
             url = repo
         elif tarball:
             if _is_url(tarball):
-                if registry and tarball.startswith(default_registry):
-                    url = registry + tarball[len(default_registry):]
+                # tarball url may start with the default registry in pnpm-lock.yaml which we must replace:
```

Comment:
I don't follow... why use the global default instead of local one?

Why does the comment use .org and .com domains?

Reply:
pnpm does a funny thing and sometimes adds `https://registry.npmjs.org/` (`DEFAULT_REGISTRY`) on the tarball URL and then puts the registry you asked for in the `registry` field. No idea why but we have to strip

Reply:
added some more color to the comment